src/modules/rf/rf.cpp

```diff
@@ -1094,3 +1124,220 @@ bool txSubFile(FS *fs, String filepath) {
+                for (int i = 0; i < received.Bit * 2; i++) {
+                    if (i > 0) received.data += " ";
+                    
+                    if (i % 2 == 0) sign = +1;
+                    else sign = -1;
+                    
+                    received.data += String(sign * (int)raw[i]);
+                }
+            
+                const char* b = dec2binWzerofill(received.key, received.Bit);
+                decimalToHexString(received.key,hexString);
+                drawMainBorder();
+                tft.setCursor(10, 28);
+                tft.setTextSize(FP);
+                tft.println("Key: " + String(hexString));
+                tft.setCursor(10, tft.getCursorY());
+                if (RfModule == 1) {
+                    tft.println("Rssi: " + String(ELECHOUSE_cc1101.getRssi()));
+                    tft.setCursor(10, tft.getCursorY());
+                }
+                tft.println("Binary: " + String(b));
+                tft.setCursor(10, tft.getCursorY());
+                tft.println("Lenght: " + String(received.Bit) + " bits");
+                tft.setCursor(10, tft.getCursorY());
+                tft.println("PulseLenght: " + String(received.te) + "ms");
+                tft.setCursor(10, tft.getCursorY());
+                tft.println("Protocol: " + String(received.protocol));
+                tft.setCursor(10, tft.getCursorY());
+                tft.println("Frequency: " + String(frequency) + " MHz");
+                tft.setCursor(10, tft.getCursorY()+LH*2);
+                tft.println("Press [NEXT] for options.");
+            }
+                
+            rcswitch.resetAvailable();
+        }
+
+        if (checkNextPress()) {
+            int option = 0;
+            if (found_freq) {
+                options = {
+                    { "Range",      [&]()  { option = 1; } },
+                    { "Signal",     [&]()  { option = 2; } },
+                };
+
+                delay(200);
+                loopOptions(options);
+            }
+            else {
+                option = 1;
+            }
+
+            if (option == 1) {
+                options = {
+                    { String("Fxd [" + String(RfFreq) + "]").c_str(), [&]()  { RfFxdFreq = 1; } },
+                    { sz_range[0], [&]()  { RfScanRange = 0; RfFxdFreq = 0; } },
+                    { sz_range[1], [&]()  { RfScanRange = 1; RfFxdFreq = 0; } },
+                    { sz_range[2], [&]()  { RfScanRange = 2; RfFxdFreq = 0; } },
+                    { sz_range[3], [&]()  { RfScanRange = 3; RfFxdFreq = 0; } },
+                };
+
+                delay(200);
+                loopOptions(options);
+
+                if (RfFxdFreq) {
+                    displayRedStripe("Scan freq set to " + String(RfFreq), TFT_WHITE, FGCOLOR);
+                }
+                else {
+                    displayRedStripe("Range set to " + String(sz_range[RfScanRange]), TFT_WHITE, FGCOLOR);
```

Comment:
after this line, add:
delay(1500);
goto RestartScan;

so we dont get trapped in the screen..